### DISCUSSION TITLE
What are de default ports used by the various components?

### DISCUSSION THREAD

**Discussion by Flightkick** [uxbox/uxbox#3975] [Q&A]
I'm trying to launch a self-hosted penpot instance in my homelab cluster.
Since the environment differs a bit from the default docker-compose set-up, I need to tweak a few things, mainly how the containers are able to communicate with each other.
The documentation does describe how to override connection parameters through the use of environment variables, but I'm uncertain which ports are being listened on by the applications inside the containers.

E.g.:

`exporter` has this `PENPOT_PUBLIC_URI` example [^1], but I'd like to know the port it's listening on as well.
`backend` has an option to configure the listening port through `PENPOT_HTTP_SERVER_PORT` [^2] and shows `6060` in the example, but is this also the default port?
`frontend` has another example for overriding the backend and exporter URI. [^3] Those do not include port numbers, so from that config alone I'd assume it would be using the default HTTP port 80. I'm a bit skeptical of that.

I could run some tests to see what is listening but since I'm not working from a stable set-up, I'd be thankful if someone with more experience with penpot could clarify things. Might also be worth documenting (in case I haven't missed it).

[^1]: https://help.penpot.app/technical-guide/configuration/#exporter
[^2]: https://help.penpot.app/technical-guide/configuration/#http
[^3]: https://help.penpot.app/technical-guide/configuration/#frontend

**Comment by Flightkick** [comment]
Figured I could answer my own question by spinning up the docker environment, derp.

Frontend: `80/tcp`
Backend: `6060/tcp`
Exporter: `6061/tcp`
Mailcatch: `1080/tcp`
Postgres: `5432/tcp`
Redis: `6379/tcp`

**Reply by AshrAF1151** [reply]
can the backend run with the swagger?
